### PR TITLE
Add typos tool to CI to automate typo detection

### PR DESCRIPTION
**Pull request by goofylfg**
### Description

- `advices` -> `advice` but It has appeared many times,Fear of side effects from excessive changes, this PR do not process this typo for now
- `extend-exclude = ["integration-tests/contracts"]`  is third party lib

### Issue Link

close:https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1781

### CODE REVIEW

**Comment by goofylfg** [comment]
@ChihChengLiang plz check it, further configuration adjustments may be required

File: .github/workflows/typos.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  typos:
+    name: Spell Check with Typos
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Use typos with config file
+        uses: crate-ci/typos@v1.21.0
+        with:
+          config: config/typos.toml
```
Comment (ChihChengLiang) [inline .github/workflows/typos.yml:20]: The CI is failing for not able to find the config. Can you try this? 
```suggestion
          config: .github/workflows/config/typos.toml
```

**Review by ChihChengLiang — CHANGES_REQUESTED** [review]

File: .github/workflows/typos.yml (hunk 1 above)
Comment (goofylfg) [reply .github/workflows/typos.yml]: > The CI is failing for not able to find the config. Can you try this?

done

**Comment by goofylfg** [comment]
@ChihChengLiang  before that have some CI failed, I updated it ,plz check it

**Review by ChihChengLiang — APPROVED** [review]
LGTM. Thank you @goofylfg for adding the typo check.

**Pull request merged**